store-implementation/simple-federated-store/src/test/java/uk/gov/gchq/gaffer/federated/simple/merge/DefaultResultAccumulatorTest.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2024 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.federated.simple.merge;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import uk.gov.gchq.gaffer.commonutil.StreamUtil;
+import uk.gov.gchq.gaffer.data.element.Entity;
+import uk.gov.gchq.gaffer.data.element.Properties;
+import uk.gov.gchq.gaffer.store.schema.Schema;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class DefaultResultAccumulatorTest {
+
+    static Stream<Arguments> commonDataArgs() {
+        return Stream.of(
+            Arguments.of(1, 2, 3),
+            Arguments.of("s1", "s2", "s1,s2"),
+            Arguments.of(true, false, false),
+            Arguments.of(true, true, true),
+            Arguments.of(
+                new ArrayList<>(Arrays.asList("a", "b")),
+                new ArrayList<>(Arrays.asList("c", "d")),
+                Arrays.asList("a", "b", "c", "d")));
+    }
+
+    @DisplayName("Should provide default merge methods for common data types")
+    @ParameterizedTest
+    @MethodSource("commonDataArgs")
+    <T> void shouldAccumulatePrimitiveDataByDefault(T result1, T result2, T expected) {
+        FederatedResultAccumulator<T> accumulator = new DefaultResultAccumulator<>();
+        assertThat(accumulator.apply(result1, result2)).isEqualTo(expected);
+    }
+
+    @Test
+    void shouldNotDoElementAggregationByDefault() {
+        Iterable<Entity> iter1 = () -> Arrays.asList(new Entity("person", "1")).iterator();
+        Iterable<Entity> iter2 = () -> Arrays.asList(new Entity("person", "1")).iterator();
+        Iterable<Entity> expected = () -> Arrays.asList(
+            new Entity("person", "1"),
+            new Entity("person", "1")).iterator();
+
+        FederatedResultAccumulator<Iterable<Entity>> accumulator = new DefaultResultAccumulator<>();
+        assertThat(accumulator.apply(iter1, iter2)).containsExactlyElementsOf(expected);
+    }
+
+    @Test
+    void shouldProvideElementAggregationBySchema() {
```

Comment:
Would be good to bulk out these tests to show how it doesn't merge 2 elements that are different and another where the graphs have multiple elements each that may or may not be merged but appreciate this is just the PoC

Reply:
Added an extra one but I agree these will need fleshing out.